### PR TITLE
[UR] CUDA multi device ctx

### PR DESCRIPTION
**Pull request by hdelan**
https://github.com/oneapi-src/unified-runtime/pull/1565

For UR multi device context, buffer interop is now deprecated since a buffer refers to multiple device pointers.

### CODE REVIEW

**Comment by hdelan** [comment]
Ping @intel/bindless-images-reviewers for review

**Review by isaacault — APPROVED** [review]

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -114,8 +114,8 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
   )
 
   fetch_adapter_source(cuda
-    ${UNIFIED_RUNTIME_REPO}
-    ${UNIFIED_RUNTIME_TAG}
+    https://github.com/hdelan/unified-runtime.git
+    cuda-multi-dev-ctx
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:118]:
```suggestion
    ${UNIFIED_RUNTIME_REPO}
    # commit d3502dc1c1eac924736fc08529e4d5f149648fd1
    # Merge: c911a9b6 71420060
    # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
    # Date:   Wed May 22 10:38:34 2024 +0100
    #     Merge pull request #1565 from hdelan/cuda-multi-dev-ctx
    #     [CUDA] CUDA adapter multi device context
    d3502dc1c1eac924736fc08529e4d5f149648fd1
```

**Review by kbenzie — COMMENTED** [review]
https://github.com/oneapi-src/unified-runtime/pull/1565 has been merged please update the UR tag as suggested

**Review by kbenzie — APPROVED** [review]

**Comment by hdelan** [comment]
@kbenzie pipeline has passed.

**Comment by hdelan** [comment]
Ping @intel/llvm-gatekeepers

**Pull request merged**